base/remote-maintenance.nix

```diff
@@ -47,9 +56,22 @@ in
       # Restrict authentication to authorized keys
       settings.PasswordAuthentication = false;
       settings.KbdInteractiveAuthentication = false;
-    };
 
-    # only with these special keys:
+      # Do not add general firewall exception
+      openFirewall = false;
+
+      # Restrict forwarding
+      extraConfig = ''
+        AllowTcpForwarding local
+        X11Forwarding no
+        AllowAgentForwarding no
+        AllowStreamLocalForwarding no
+      '';
```

Comment:
I initially hesitated, but you are right, there is little from historical experience that suggests keeping it. Let's disallow.

Reply:
I looked into it more, and now would again say to keep allowing it.

1. This may actually be useful for some applications. Have used `scp` for production reports, for example.

2. Disallowing SFTP is not effective, because legacy scp protocol may still be used (`scp -O` on client) and is hard to rule out while still allowing shell access.
  
  ```
  # client
  $ oh hai > my-little-file 
  $ scp -O ./my-little-file root@10.147.20.240:/home/play/legacy-ftw
  my-little-file                    100%    7     2.6KB/s   00:00
  # playos host with `allowSFTP = false`
  [root@playos-test:~]# cat /home/play/legacy-ftw 
  oh hai
  ```

Hardening access/authentication mechanisms seems more beneficial, and limiting to the ZeroTier interface already seems like a big gain in this respect.